### PR TITLE
image import workflow PVS-to-PVC image import

### PR DESCRIPTION
**Pull request by iv1111**

### CODE REVIEW

**Comment by iv1111** [comment]
creating this PR but the code is not ready yet and needs to be extended and refactored

**Comment by iv1111** [comment]
@agrare I've removed the WIP. Can we get this one merged? One thing missing only is the ImportDispatcher implementation which I'll implement in a separate/next PR.

**Comment by iv1111** [comment]
@miq-bot assign-label najdorf/yes?

**Comment by iv1111** [comment]
@miq-bot add_label najdorf/yes?

**Comment by iv1111** [comment]
@jaywcarman FYI.

File: app/models/manageiq/providers/ibm_power_vc/cloud_manager/template.rb [hunk 1]
```diff
@@ -6,4 +6,101 @@ class ManageIQ::Providers::IbmPowerVc::CloudManager::Template < ManageIQ::Provid
...
+      :session_id      => session_id,
+      :ems_id          => ext_management_system.id,
+      :src_provider_id => options['src_provider_id'].to_i,
+      :cos_id          => options['obj_storage_id'],
+      :miq_img         => get_image_info(options['src_image_id'])[:name],
+      :img_id          => MiqTemplate.find(options['src_image_id']).ems_ref,
+      :cos_data        => cos_data,
+      :import_creds_id => import_creds,
+      :ssh_creds_id    => ssh_creds,
+      :playbook_path   => ManageIQ::Providers::IbmPowerVc::Engine.root.join("content/ansible_runner/run.yml"),
```
Comment (agrare) [inline app/models/manageiq/providers/ibm_power_vc/cloud_manager/template.rb:60]: I think this needs a more descriptive name than `run.yml` maybe `image-copy-pvs-to-pvc.yml` to match the role name?

File: app/models/manageiq/providers/ibm_power_vc/cloud_manager/template.rb [hunk 2]
```diff
@@ -6,4 +6,101 @@ class ManageIQ::Providers::IbmPowerVc::CloudManager::Template < ManageIQ::Provid
...
+
+    key  = Base64.strict_encode64(cipher.random_key)
+    iv   = Base64.strict_encode64(cipher.random_iv)
+    encr = Base64.strict_encode64(cipher.update(creds.to_json) + cipher.final)
+
+    [encr, key, iv]
+  end
+
+  private_class_method def self.image_ems_ref(bucket_id)
+    MiqTemplate.find(bucket_id).uid_ems
```
Comment (agrare) [inline app/models/manageiq/providers/ibm_power_vc/cloud_manager/template.rb:104]: Is `bucket_id` the best var for a template?  I think of `CloudObjectStoreContainer` when I hear `bucket` not an image, maybe `image_id` ?

File: app/models/manageiq/providers/ibm_power_vc/cloud_manager/template.rb (hunk 1 above)
Comment (iv1111) [reply app/models/manageiq/providers/ibm_power_vc/cloud_manager/template.rb]: done

File: app/models/manageiq/providers/ibm_power_vc/cloud_manager/template.rb [hunk 3]
```diff
@@ -6,4 +6,101 @@ class ManageIQ::Providers::IbmPowerVc::CloudManager::Template < ManageIQ::Provid
...
+  extend ActiveSupport::Concern
+
+  supports :import_image
+
+  def self.raw_import_image(ext_management_system, options = {})
+    session_id = SecureRandom.uuid
+    wrkfl_timeout = options['timeout'].to_i.hours
+
+    cos = ExtManagementSystem.find(options['obj_storage_id'])
+    raise MiqException::Error, _("unable to find cloud object storage by this id '#{options['obj_storage_id']}'") if cos.nil?
```
[the remark this answers is not in the document]
Comment (iv1111) [reply app/models/manageiq/providers/ibm_power_vc/cloud_manager/template.rb]: done

File: app/models/manageiq/providers/ibm_power_vc/cloud_manager/template.rb (hunk 2 above)
Comment (iv1111) [reply app/models/manageiq/providers/ibm_power_vc/cloud_manager/template.rb]: done

**Review by agrare — APPROVED** [review]

**Pull request merged**